### PR TITLE
JDK-8263559: Add missing initializers to VM_PopulateDumpSharedSpace

### PR DESCRIPTION
**Pull request by tstuefe**
Trivial.

Sonarcloud reports missing initializers in VM_PopulateDumpSharedSpace. I don't see how this could lead to an error, since these members are initialized later, but for cleanliness sake lets initialize them.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263559](https://bugs.openjdk.java.net/browse/JDK-8263559): Add missing initializers to VM_PopulateDumpSharedSpace

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to fe44ca6467f169c76bc4c03a2e42c718c6421fd3
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to fe44ca6467f169c76bc4c03a2e42c718c6421fd3
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to fe44ca6467f169c76bc4c03a2e42c718c6421fd3

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2996/head:pull/2996`
`$ git checkout pull/2996`

### CODE REVIEW

**Review by kimbarrett — COMMENTED** [review]
It seems like the later assignments to NULL should no longer be needed, and maybe just confuse things by making the reader wonder if they got assigned to something else at some point between construction and the assignments.

**Review by iklam — APPROVED** [review]
Looks good and trivial. Thanks for fixing this.

**Review by iklam — CHANGES_REQUESTED** [review]
Oops,  agree with Kim. The later assignments to NULL should be removed.

**Comment by tstuefe** [comment]
Kim, Ioi, thanks. I removed the explicit NULL assignments.

Side note, I know VM_op::do_it() is supposed to run only once, and in straight succession from constructing the object, but I wonder if it would make sense to assert this.

**Review by iklam — APPROVED** [review]
LGTM

File: src/hotspot/share/memory/metaspaceShared.cpp [hunk 1]
```diff
@@ -386,8 +386,12 @@ class VM_PopulateDumpSharedSpace : public VM_GC_Operation {
...
 
-  VM_PopulateDumpSharedSpace() : VM_GC_Operation(0, /* total collections, ignored */
-                                                 GCCause::_archive_time_gc)
+  VM_PopulateDumpSharedSpace() :
+    VM_GC_Operation(0 /* total collections, ignored */, GCCause::_archive_time_gc),
+    _closed_archive_heap_regions(NULL),
+    _open_archive_heap_regions(NULL),
+    _closed_archive_heap_oopmaps(NULL),
+    _open_archive_heap_oopmaps(NULL)
   { }
```
Comment (coleenp) [inline src/hotspot/share/memory/metaspaceShared.cpp:395]: The { } should go on line 394, since { is always on the declaration line.

**Review by coleenp — APPROVED** [review]
LGTM!

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by tstuefe** [comment]
Thanks Coleen, Kim and Ioi!

@coleenp I moved the constructor braces up. 

/integrate

**Pull request closed without merging**